File: Sc.Models/Dtos/ReceivedVouchers/Base/BaseReceivedVoucherDto.cs
using Sc.Models.Dtos.Companies;
using Sc.Models.Dtos.Suppliers;
using Sc.Models.Entities.Base;
using Sc.Models.Enums.ReceivedVouchers;

namespace Sc.Models.Dtos.ReceivedVouchers.Base
{
    public abstract class BaseReceivedVoucherDto : Entity
    {
        public DateTime ContractDate { get; set; }
        public string ContractNumber { get; set; }

        public ReceivedVoucherState State { get; set; }

        public int CompanyUserId { get; set; }
        public int CompanyId { get; set; }
        public CompanyDto Company { get; set; }

        public int? SupplierId { get; set; }
        public SupplierDto Supplier { get; set; }
        public int? OfferingId { get; set; }
        public SupplierOfferingDto Offering { get; set; }
        public bool OfferingAdditionalPayment { get; set; }
        public string ReceivedOffering { get; set; }
        public string OfferingClarifications { get; set; }

        public int? SecondSupplierId { get; set; }
        public SupplierDto SecondSupplier { get; set; }
        public int? SecondOfferingId { get; set; }
        public SupplierOfferingDto SecondOffering { get; set; }
        public bool SecondOfferingAdditionalPayment { get; set; }
        public string SecondReceivedOffering { get; set; }
        public string SecondOfferingClarifications { get; set; }
    }
}

File: WebApplication/src/received-vouchers/dtos/base/base-received-voucher.dto.ts
import { CompanyDto } from "src/companies/dtos/company.dto";
import { ReceivedVoucherState } from "src/received-vouchers/enums/received-voucher-state.enum";
import { SupplierOfferingDto } from "src/suppliers/dtos/supplier-offering.dto";
import { SupplierDto } from "src/suppliers/dtos/supplier.dto";

export class BaseReceivedVoucherDto {
    id: number;

    contractDate: Date;
    contractNumber: string;

    state: ReceivedVoucherState;

    companyUserId: number;
    companyId: number;
    company: CompanyDto;

    supplierId: number;
    supplier: SupplierDto;
    offeringId: number;
    offering: SupplierOfferingDto;
    offeringAdditionalPayment: boolean;
    receivedOffering: string;
    offeringClarifications: string;

    secondSupplierId: number;
    secondSupplier: SupplierDto;
    secondOfferingId: number;
    secondOffering: SupplierOfferingDto;
    secondOfferingAdditionalPayment: boolean;
    secondReceivedOffering: string;
    secondOfferingClarifications: string;
}